### PR TITLE
Go-friendly interface, hide implementation details.

### PR DESCRIPTION
**Pull request by marcinwyszynski**

### CODE REVIEW

File: spf2.go [hunk 1]
```diff
@@ -21,40 +22,63 @@ const (
...
 	s *C.SPF_server_t
 }
 
-// NewServer creates a new server
-func NewServer() *Server {
-	srv := new(Server)
-	srv.s = C.SPF_server_new(C.SPF_DNS_CACHE, 0)
-	return srv
+// NewClient creates a new SPF client.
+func NewClient() Client {
```
Comment (eaigner) [inline spf2.go:35]: Why did you rename the `Server` to `Client`? It obviously uses the `C.SPF_server_new` function.

Comment (marcinwyszynski) [inline spf2.go:35]: Yes, it does. It's a rather unfortunate name for what it does though.

**Comment by porjo** [comment]
I like the idea of an all-in-one convenience method , however I think the other methods should still be public.
